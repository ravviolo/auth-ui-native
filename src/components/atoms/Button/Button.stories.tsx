import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { StoryScreen } from 'storybook/StoryScreen';

import { Button } from './Button';

storiesOf('Button', module)
  .addDecorator((getStory) => <StoryScreen>{getStory()}</StoryScreen>)
  .add('Default', () => (
    <Button
      testID={text('testID', 'test-id')}
      title={text('title', 'Button')}
      onPress={action('onPress')}
    />
  ));
