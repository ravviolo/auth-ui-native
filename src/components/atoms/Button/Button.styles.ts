import { StyleSheet } from 'react-native';
import { colors } from 'theme';

export const styles = StyleSheet.create({
  btnContainer: {
    alignItems: 'center',
    backgroundColor: colors.primary,
    borderRadius: 10,
    elevation: 5,
    padding: 15,
  },
  btnText: {
    color: colors.textWhite,
    letterSpacing: 1,
    textTransform: 'uppercase',
  },
});
