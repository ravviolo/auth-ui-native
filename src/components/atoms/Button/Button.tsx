import { StyleSheet, Text, TouchableHighlight } from 'react-native';
import { colors } from 'theme';

export interface Props {
  title: string;
  testID: string;
  onPress: () => void;
}

export const Button = ({ onPress, testID, title }: Props) => {
  return (
    <TouchableHighlight
      style={styles.btnContainer}
      testID={testID}
      underlayColor={colors.primaryDark}
      onPress={onPress}
    >
      <Text style={styles.btnText}>{title}</Text>
    </TouchableHighlight>
  );
};

const styles = StyleSheet.create({
  btnContainer: {
    alignItems: 'center',
    backgroundColor: colors.primary,
    borderRadius: 10,
    elevation: 5,
    padding: 15,
  },
  btnText: {
    color: colors.textWhite,
    letterSpacing: 1,
    textTransform: 'uppercase',
  },
});
