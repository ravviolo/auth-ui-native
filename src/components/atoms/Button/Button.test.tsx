/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import { fireEvent, render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { Button } from './Button';

import type { Props } from './Button';

describe('Button', () => {
  const mockOnPress = jest.fn<Props['onPress'], []>();
  const props: Props = {
    testID: 'btn-test-id',
    title: 'Test',
    onPress: mockOnPress,
  };

  it('should display text received via props', () => {
    render(<Button {...props} />);

    const btn = screen.getByTestId('btn-test-id');

    expect(btn).toHaveTextContent('Test');
  });

  it('should handle click event, fires event handler once', () => {
    render(<Button {...props} />);

    const btn = screen.getByTestId('btn-test-id');

    fireEvent.press(btn);

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    const tree = testRenderer.create(<Button {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
