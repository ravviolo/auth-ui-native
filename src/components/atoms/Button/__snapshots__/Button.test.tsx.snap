// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FF4684",
      "borderRadius": 10,
      "elevation": 5,
      "padding": 15,
    }
  }
  testID="btn-test-id"
>
  <Text
    style={
      Object {
        "color": "#EDEDED",
        "letterSpacing": 1,
        "textTransform": "uppercase",
      }
    }
  >
    Test
  </Text>
</View>
`;
